File: 09.BinarySearch/T1560.2.java
// 测试平台: https://nanti.jisuanke.com/t/T1560
import java.io.*;
import java.util.*;

public class Main {
  static private int lowerBound(long[] A, int n, long target) {
    int l = 0, r = n;
    while (l < r) {
      final int m = l + ((r - l) >> 1);
      if (A[m] < target) {
        l = m + 1;
      } else {
        r = m;
      }
    }
    return l;
  }

  static private boolean binarySearch(long[] A, int n, long target) {
    int l = lowerBound(A, n, target);
    return l < n && A[l] == target;
  }

  public static void main(String args[]) throws Exception {
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int m = sc.nextInt();
    long[] A = new long[n];

    for (int i = 0; i < n; i++) {
      A[i] = sc.nextLong();
    }
    Arrays.sort(A);
    for (int i = 0; i < m; i++) {
      long target = sc.nextLong();
      System.out.println(binarySearch(A, n, target) ? "YES" : "NO");
    }
  }
}

File: 09.BinarySearch/T1560.2.py
# -*- coding: utf-8 -*-

# 测试平台: https://nanti.jisuanke.com/t/T1560

def lowerBound(A, target):
    l = 0
    r = len(A) if A else 0
    while l < r:
        m = l + ((r-l)>>1)
        if (A[m] < target):
            l = m + 1
        else:
            r = m
    return l

def binarySearch(A, target):
    l = lowerBound(A, target)
    return True if l < len(A) and A[l] == target else False

s = raw_input().split()
n = int(s[0])
m = int(s[1])
s = raw_input().split()
A = [int(x) for x in s]
A.sort()

for i in range(0, m):
    x = int(raw_input())
    print "%s" % ("YES" if binarySearch(A, x) else "NO")


